### PR TITLE
Migrate some nft ethereum spells - Ready for review 

### PR DESCRIPTION
**Pull request by henrystats**
# Thank you for contributing to Spellbook!
Please refer to the top of the `readme` in the root of Spellbook to learn how to contribute to Spellbook on DuneSQL.

### CODE REVIEW

File: models/nft/ethereum/nft_ethereum_collection_stats.sql [hunk 1]
```diff
@@ -17,18 +19,18 @@ WITH src_data as
...
         , currency_symbol
         , amount_original
         , amount_usd
     FROM 
         {{ ref('nft_trades') }} 
     WHERE blockchain = 'ethereum'
-        AND number_of_items = 1
-        AND tx_from != LOWER('0x0000000000000000000000000000000000000000')
-        AND amount_raw > 0
+        AND CAST(number_of_items as DOUBLE) = 1
```
Comment (0xRobin) [inline models/nft/ethereum/nft_ethereum_collection_stats.sql:29]:
```suggestion
        AND number_of_items = UINT256 '1'
```

File: models/nft/ethereum/nft_ethereum_top_sales.sql [hunk 2]
```diff
@@ -23,7 +25,7 @@ WITH src AS
         blockchain = 'ethereum'
         AND currency_symbol IN ('ETH', 'WETH')
         AND amount_original IS NOT NULL
-        AND number_of_items = 1
+        AND CAST(number_of_items as double) = 1
```
Comment (0xRobin) [inline models/nft/ethereum/nft_ethereum_top_sales.sql:28]:
```suggestion
        AND number_of_items = UINT256 '1'
```

File: models/nft/ethereum/nft_ethereum_wallet_pnl.sql [hunk 3]
```diff
@@ -20,11 +22,11 @@ WITH weekly_unique_wallet_address as
     FROM 
         {{ ref('nft_trades') }}
     WHERE 
-        block_time >= date_trunc("day", now() - interval '1 week')
+        block_time >= date_trunc('day', now() - interval '7' Day)
         AND blockchain = 'ethereum'
         AND currency_symbol IN ('WETH', 'ETH')
         AND amount_original IS NOT NULL
-        AND number_of_items = 1
+        AND CAST(number_of_items as DOUBLE) = 1
```
Comment (0xRobin) [inline models/nft/ethereum/nft_ethereum_wallet_pnl.sql:29]:
```suggestion
        AND number_of_items = UINT256 '1'
```

File: models/nft/ethereum/nft_ethereum_wallet_pnl.sql [hunk 4]
```diff
@@ -35,11 +37,11 @@ WITH weekly_unique_wallet_address as
...
```
Comment (0xRobin) [inline models/nft/ethereum/nft_ethereum_wallet_pnl.sql:44]:
```suggestion
        AND number_of_items = UINT256 '1'
```

File: models/nft/ethereum/nft_ethereum_wallet_pnl.sql [hunk 5]
```diff
@@ -62,7 +64,7 @@ WITH weekly_unique_wallet_address as
         src.currency_symbol IN ('ETH', 'WETH')
         AND src.blockchain = 'ethereum'
         AND src.buyer != src.seller 
-        AND src.number_of_items = 1
+        AND CAST(src.number_of_items as DOUBLE) = 1
```
Comment (0xRobin) [inline models/nft/ethereum/nft_ethereum_wallet_pnl.sql:67]:
```suggestion
        AND src.number_of_items = UINT256 '1'
```

File: models/nft/ethereum/nft_ethereum_wallet_pnl.sql [hunk 6]
```diff
@@ -87,7 +89,7 @@ WITH weekly_unique_wallet_address as
...
```
Comment (0xRobin) [inline models/nft/ethereum/nft_ethereum_wallet_pnl.sql:92]:
```suggestion
        AND src.number_of_items = UINT256 '1'
```

File: models/nft/ethereum/nft_ethereum_wallet_pnl.sql [hunk 7]
```diff
@@ -143,7 +145,7 @@ WITH trades as
...
```
Comment (0xRobin) [inline models/nft/ethereum/nft_ethereum_wallet_pnl.sql:148]:
```suggestion
        AND src.number_of_items = UINT256 '1'
```

File: models/nft/ethereum/nft_ethereum_wallet_pnl.sql [hunk 8]
```diff
@@ -164,7 +166,7 @@ WITH trades as
...
```
Comment (0xRobin) [inline models/nft/ethereum/nft_ethereum_wallet_pnl.sql:169]:
```suggestion
        AND src.number_of_items = UINT256 '1'
```

**Review by 0xRobin — CHANGES_REQUESTED** [review]
1 minor comment about the `number_of_items` comparison.

**Comment by henrystats** [comment]
thank you ser I'll update shortly @0xRobin, the test is failing because the data is different from the spell & the test. Not sure who set up the tests (so I could check against their sources).

But I'll verify against nft.trades data and ensure that the query runs correctly and that the figures are correct from nft.trades at least.

**Comment by 0xRobin** [comment]
> Not sure who set up the tests (so I could check against their sources).

You set up the test. :smile: 
Note that ` approx_percentile` might produce some different results than the spark equivalent. It is also not deterministic. (the same run might not always result in the exact same result)

**Comment by henrystats** [comment]
thank you ser @0xRobin this is now ready for review... I actually don't/didn't remember writing this particular test for collection stats, I remembered writing the spells but I must have missed the tests somehow 😆 so I'm thinking someone else wrote them.

At the time the test was written, some trades weren't included in nft.trades (not sure if it was specific projects trades missing or the project had not been added to nft.trades). I double checked the data and trades and saw that they were correct so I updated the test target values.

**Review by 0xRobin — APPROVED** [review]
great work!

**Comment by 0xRobin** [comment]
@jeff-dude @antonio-mendes @couralex6 let's merge this in.

**Pull request merged**